lib/capability/capability.go

```diff
@@ -22,11 +24,18 @@ func (c *ClusterCapabilities) Equal(capabilities *ClusterCapabilities) error {
 	return nil
 }
 
-type capabilitiesSort []configv1.ClusterVersionCapability
-
-func (caps capabilitiesSort) Len() int           { return len(caps) }
-func (caps capabilitiesSort) Swap(i, j int)      { caps[i], caps[j] = caps[j], caps[i] }
-func (caps capabilitiesSort) Less(i, j int) bool { return string(caps[i]) < string(caps[j]) }
+// SortedList returns the slice with contents in sorted order for a given set of configv1.ClusterVersionCapability.
+// It returns nil if the give set is nil.
+func SortedList(s sets.Set[configv1.ClusterVersionCapability]) []configv1.ClusterVersionCapability {
+	if s == nil {
+		return nil
+	}
+	l := s.UnsortedList()
+	sort.Slice(l, func(i, j int) bool {
+		return l[i] < l[j]
+	})
+	return l
+}
```

Comment:
Why cant we use https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets#List ?

Reply:
:TIL

I couldn't imagine `List` is defined in the `sets` lib. 😛 

Replaced the sorting with `List()`. 

We still need the nil checking as `List()` returns the empty set instead of nil which would lead to many errors on unit tests. I tried to fix them, such as replacing `reflect.DeepEqual()` with `cmp.Diff()` + `ignores` etc (the code gets ugly quickly).

Reply:
Yeah the treatment of `nil` / empty set is a bit awkward, I'm not a fan of the pattern that's now everywhere:

```go
if caps.Len() == 0 {
	return nil
}
return caps
```

But I guess we can tackle this as an increment if we want to - removing `SortedList` would be the first step and anyone interested can just follow the compiler.